rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2245,6 +2245,58 @@ namespace rsx
 		}
 	}
 
+	void thread::fifo_wake_delay(u64 div)
+	{
+		// TODO: Nanoseconds accuracy
+		u64 remaining = g_cfg.video.driver_wakeup_delay;
+
+		if (!remaining)
+		{
+			return;
+		}
+
+		// Some cases do not need full delay
+		remaining = ::aligned_div(remaining, div);
+		const u64 until = get_system_time() + remaining;
+
+		// TODO: Determine best value 
+		constexpr u32 try_yield_delay = 2;
```

Comment:
This needs to go up a bit. Yield penalty can be pretty large even when there is no thread contention, you can read the recent rants about it online.

Reply:
I suggest ignoring yield for anything smaller than host_min_quantum / 2

Reply:
dunno, it probably needs to be adjusted differently for Windows and Linux. Need precise values for each platform, I doubt host_min_quantum / 2 is the correct value and might be too high for atleast one platform.
I mean its used with relatively good accuracy in sys_timer_usleep.

Reply:
I will use host_min_quantum / 3 just to be sure for now

Reply:
The reason it "works well" for usleep is because the caller is already willingly reliquinshing control which means they're already outside a critical code section. Even if you go overboard its not really a big deal in that case. .e g For RSX, when state is idle, yield may be preferred, but when waiting for a jump-to-self spinlock to clear, not so much unless the system is significantly resource starved like a dual core or something.